plugins/guests/linux/cap/public_key.rb

```diff
@@ -53,6 +53,7 @@ def self.remove_public_key(machine, contents)
               result=$?
             fi
             rm -f '#{remote_path}'
+            chmod 0600 ~/.ssh/authorized_keys
```

Comment:
It's still in this line though:

```ruby
mv ~/.ssh/authorized_keys.tmp ~/.ssh/authorized_keys && chmod 0600 ~/.ssh/authorized_keys
```

At line 56. This should still be chmod'ing the correct file. Are you finding that's not the case? Do you have any vagrant logs to share?

Thanks!